lib/sqewer/connection.rb

```diff
@@ -211,7 +211,10 @@ def handle_batch_with_retries(method, batch)
   end
 
   def client
-    @client ||= Aws::SQS::Client.new(
+    # It's better using a singleton client to prevent making a lot of HTTP
+    # requests to the AWS metadata endpoint when getting credentials.
+    # Maybe in the future, we can remove @client and use Storm.client only.
+    Sqewer.client || @client ||= Aws::SQS::Client.new(
```

Comment:
Can we unpack this into ifs/elses? It is a bit hard for me to understand what the priorities of comparison will be

Reply:
you're right! I had to test it manually before 😄 

I changed in 5e486f9